File: hackerrank/warmup/birthday_candles.rb
# https://www.hackerrank.com/challenges/birthday-cake-candles?h_r=next-challenge&h_v=zen

def birthdayCakeCandles(n, ar)
    # Complete this function
    max = ar.max
    ar.select {|x| x == max}.count
end

n = gets.strip.to_i
ar = gets.strip
ar = ar.split(' ').map(&:to_i)
result = birthdayCakeCandles(n, ar)
puts result;

File: hackerrank/warmup/birthdayCandles.js
// https://www.hackerrank.com/challenges/birthday-cake-candles?h_r=next-challenge&h_v=zen

function birthdayCakeCandles(arr) {
  // let max = arr.reduce(function(a,b) {
  //   return Math.max(a,b);
  // })
  let max = arr[0];
  let count = 0;
  for (let i = 0; i < arr.length; i++) {
    if (arr[i] > max) {
      max = arr[i];
      count = 1
    } else if (arr[i] === max) {
      count += 1;
    }
  }
  return count;
}
